app/routes/project/messages.js

```diff
@@ -0,0 +1,45 @@
+import { get } from '@ember/object';
+import { inject as service } from '@ember/service';
+import Route from '@ember/routing/route';
+import AuthenticatedRouteMixin from 'ember-simple-auth/mixins/authenticated-route-mixin';
+import { CanMixin } from 'ember-can';
+
+export default Route.extend(AuthenticatedRouteMixin, CanMixin, {
+  session: service(),
+  store: service(),
+
+  beforeModel() {
+    // we need to deal with overriding the `AuthenticatedRouteMixin`
+    let isAuthenticated = get(this, 'session.isAuthenticated');
+
+    if (isAuthenticated) {
+      return this._ensureUserHasCredentials(...arguments);
+    } else {
+      // call `beforeModel` in `AuthenticatedRouteMixin`
+      return this._super(...arguments);
+    }
+  },
+
+  async model() {
+    console.log('model');
+    let projectId = get(this.modelFor('project'), 'id');
+    let store = get(this, 'store');
+    // peekAll returns a live array, so it will alse render any newly created
+    // messages, causing the list to update properly on project.messages.new
+    await store.query('conversation', { project_id: projectId });
+    return store.peekAll('conversation');
```

Comment:
We have a `list-detail` UI here

- a list of conversations in the sidebar on `/messages/`
- clicking one of the items navigates to `/messages/:id`, and renders the sidebar in the outlet of the main area, while still displaying the list
- navigating to `/messages/new` does the same

This model hook allows the store to detect a new conversation record and show it in the list (styled differently). 

It's hacky, but it's the only way I could find to do this.

An alternative would be to clone the query result of the `project.messages` model into an array, then render that array in controller, instead of the query result directly. 

That would allow us to push the new conversation into that array during the `messages.new` `afterModel` hook. 

Pushing into the query result directly, which is what I initially tried, does not seem to be possible without resorting to using private functions and properties of `DS.model`